src/components/PlanCard/PlanCard.tsx

```diff
@@ -49,6 +50,16 @@ export const PlanCard = ({
       <PlanNav>
         <PillLabel type="primary"> test label</PillLabel>
         <PillLabel type="secondary"> test label</PillLabel>
+        <PrimaryButton onClick={handleClick}>
+          Share
+          <SharePlanModal
+            className={''}
+            classID={''}
+            isOpen={true}
+            toggle={toggle}
```

Comment:
you should use `isOpen` and `toggle` from the hook
```
const { isOpen, toggle } = useModal();
```

The way it works is that, `useModal` is a hook that you can import and use in the React component, and it returns `isOpen` state and `toggle` state setter.
  - `isOpen` is the state of the modal, being opened or not (true/false)
  - `toggle` is a function that will switch the value of `isOpen` to true/false

You will need `isOpen` to let `Modal` know if the modal should be shown or not. And you need to pass down `toggle` to the close or click handler in the modal, in order to trigger the state `isOpen` to be updated.

Reply:
I don't understand this part. Can you show me sometime, maybe after one of our standup meetings next week?